classes/class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -67,7 +64,7 @@ public function run( $request ) {
 				'tracking' => $label_data->label->tracking_id,
 				'refundable_amount' => $label_data->label->refundable_amount,
 				'created' => $label_data->label->created,
-				'carrier_id' => $settings[ 'carrier' ],
+				'carrier_id' => $label_data->label->carrier_id,
```

Comment:
Good catch! Fixed in https://github.com/Automattic/woocommerce-connect-server/pull/879